iquip/apps/scheduler.py

```diff
@@ -223,6 +258,39 @@ def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:
         return experimentView.sizeHint()
 
 
+class SchedulerPostWorker(QObject):
+    """Worker for posting a request to the proxy server, targeting the scheduler.
+
+    Signals:
+        done: The signal is emitted when the procedure of the worker is done.
+
+    Attributes:
+        mode: The type of command that is requested to the server.
+        rid: The run identifier value of the target experiment.
+    """
+    done = pyqtSignal()
+
+    def __init__(self, mode: str, rid: int):
+        """Extended.
+
+        Args:
+            mode: The type of command that is requested to the server.
+            rid: The run identifier value of the target experiment.
+        """
+        super().__init__()
+        self.mode = mode
+        self.rid = rid
+
+    def run(self):
+        """Overridden."""
+        base_path = "http://127.0.0.1:8000/experiment/"
+        if self.mode == "delete":
+            requests.post(base_path + "delete", params={"rid": self.rid}, timeout = 10)
+        elif self.mode == "request_termination":
+            requests.post(base_path + "terminate", params={"rid": self.rid}, timeout = 10)
```

Comment:
Is there a reason you separated `mode` and the path? For example, "request_termination" and "terminate".

Reply:
So the full name of the function is `request termination` in artiq server, so I wanted to get that name preserved, but at the same time `terminate` is the URL here.

Reply:
However, the "API" is composed of the URL paths, not the artiq proxy server functions.
I think this design inserts an additional unnecessary layer (2):
```
0. user
1. -> GUI menu "Request termination"
2. -> mode "request_termination"
3. -> URL path "terminate"
4. -> server function "request_termination"
5. -> artiq scheduler
```
Isn't it more natural to match the `mode` to layer 3 ("terminate") instead of 4 ("request_terminate")?

Reply:
Is it resolved now?

Reply:
Yes, that's exactly what I thought.